Repositories/DbUserRepository.cs

```diff
@@ -25,5 +24,13 @@ public IEnumerable<Listing> GetAllRequestedListingsByUserId(int userId)
             _context.Entry(user).Collection<Listing>(user => user.RequestedListings).Load();
             return user.RequestedListings;
         }
+
+        public bool DublicateUsername(string username)
+        {
+            var user = entities.Find(username);
```

Comment:
entities.Find searches by key, which is Id, in this case.
I would suggest you to use GetAll().Any(u => u.Username == username) in UserService or UserController.
Also, you should probably check that email is different also.

Reply:
If you will have time next week, you could look into making these fields unique something like this:
![image](https://user-images.githubusercontent.com/90030313/145537475-79177f0f-f615-4d9a-bf05-69da28ab8be6.png)
And then try to catch exceptions on insertion or maybe it would be possible to use Find or some other method to seach by username or email directly in DbSet.